logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -142,6 +142,11 @@ def config_init(params)
       instance_variable_set("@#{key}", value)
     end
 
+    # now that we know the parameters are valid, we can obfuscate the original copy
+    # of the parameters before storing them as an instance variable
+    self.class.validate_check_parameter_values(original_params)
```

Comment:
I am OK with this fix event if we are actually validating twice the submitted values.
The actual problem is we should split the value assignment from the validation.